### PR TITLE
Added support for BedrockRuntime GenAI attributes

### PR DESCRIPTION
**Pull request by ezhang6811**
*Description of changes:*
Updated local OTEL repo to support the following GenAI attributes for BedrockRuntime InvokeModel calls:
`gen_ai.request.max_tokens`
`gen_ai.request.temperature`
`gen_ai.request.top_p`
`gen_ai.usage.input_tokens`
`gen_ai.usage.output_tokens`
`gen_ai.response.finish_reasons`

E2E testing below shows that these additional GenAI attributes are made visible with the InvokeModel call for each of the 6 models supported by Bedrock.
GenAI attributes are incomplete for several models because the Bedrock API for .NET does not expose the response headers in the API response, leaving only what's available in the response body for extraction. See [this doc](https://quip-amazon.com/slHRAqjxvbf6/Missing-GenAI-Attributes-in-NET-SDK) for more details.

Amazon Titan:
image
Anthropic Claude:
image
Meta Llama:
image
Cohere Command:
image
AI21 Labs Jamba:
image
Mistral AI:
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSLlmModelProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using Amazon.Runtime;
+
+namespace OpenTelemetry.Instrumentation.AWS.Implementation;
+
+internal class AWSLlmModelProcessor
+{
+    internal static void ProcessRequestModelAttributes(Activity activity, AmazonWebServiceRequest request, string model)
```
Comment (AsakerMohd) [inline src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSLlmModelProcessor.cs:13]: So there is a lot of duplicate code here that I think should be refactored. For example, `ProcessRequestModelAttributes` and `ProcessResponseModelAttributes` are the almost the same except the parameter list. Even the body extraction is the same: GetType -> GetProperty -> GetValue -> Process. Could you look on refactoring this code a bit more? Let me know what you think or if you want to discuss.

**Review by AsakerMohd — COMMENTED** [review]
So I see that you did an E2E test to verify the changes. What about continuous testing tho? I was taking a look and we don't have any unit tests for this package in our package here which was surprising. If we decide to merge this pkg with upstream down the line, they will want unit tests and some of those already exist upstream. For that tho, it's definitely out of scope for this task but we need an immediate follow up. Could you create an issue in this repo about it and then bring it up in the next scrum meeting to discuss more about it with the team?

What about contract tests tho since we already have those setup? Basically how do we guarantee that these attributes don't change/break down the line without being explicitly informed about them?

File: src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSLlmModelProcessor.cs (hunk 1 above)
Comment (ezhang6811) [reply src/OpenTelemetry.Instrumentation.AWS/Implementation/AWSLlmModelProcessor.cs]: I consolidated those two functions into one, as well as the model-specific functions. There wasn't a good base type shared by `AmazonWebServiceRequest` and `AmazonWebServiceResponse`, so I used a generic type. Let me know if approach is more preferable.

**Review by AsakerMohd — APPROVED** [review]

**Review by AsakerMohd — APPROVED** [review]

**Comment by ezhang6811** [comment]
Merging this PR with https://github.com/aws-observability/aws-otel-dotnet-instrumentation/pull/137.

**Pull request closed without merging**